json_path/src/json_node.rs

```diff
@@ -33,10 +33,10 @@ impl SelectValue for Value {
         }
     }
 
-    fn values<'a>(&'a self) -> Option<Box<dyn Iterator<Item = &'a Self> + 'a>> {
+    fn values<'a>(&'a self) -> Option<Box<dyn Iterator<Item = ValueRef<'a, Self>> + 'a>> {
         match self {
-            Self::Array(arr) => Some(Box::new(arr.iter())),
-            Self::Object(o) => Some(Box::new(o.values())),
+            Self::Array(arr) => Some(Box::new(arr.iter().map(|v| ValueRef::Borrowed(v)))),
```

Comment:
Stylistically, would it be possible to make this point-free? ie `.map(ValueRef::Borrowed)` instead of `.map(|v| ValueRef::Borrowed(v))`. It's not a big deal either way.